backend/messaging/wizard_topic.go

```diff
@@ -19,75 +21,98 @@ type wizardTopic struct {
 	bindings map[string]Subscriber
 	ring     types.Ring
 	sync.RWMutex
+	droppedMessages int64
+	done            chan struct{}
 }
 
-// Send a message to all subscribers to this topic.
-func (wTopic *wizardTopic) Send(msg interface{}) {
-	wTopic.RLock()
-	defer wTopic.RUnlock()
+func (t *wizardTopic) logDroppedMessages() {
+	dropped := atomic.LoadInt64(&t.droppedMessages)
+	if dropped == 0 {
+		return
+	}
+	// Inexact but avoids contention; good enough for our purposes.
+	atomic.StoreInt64(&t.droppedMessages, 0)
+	logger.WithFields(logrus.Fields{
+		"topic": t.id,
+		"dropped_messages_per_second": dropped}).Warn(
+		"not all messages could be delivered")
+}
 
-	for _, subscriber := range wTopic.bindings {
+// Send a message to all subscribers to this topic.
+func (t *wizardTopic) Send(msg interface{}) {
+	t.RLock()
+	names := make([]string, 0, len(t.bindings))
+	subscribers := make([]Subscriber, 0, len(t.bindings))
+	for name, subscriber := range t.bindings {
+		names = append(names, name)
+		subscribers = append(subscribers, subscriber)
+	}
+	t.RUnlock()
+	for i, subscriber := range subscribers {
 		select {
 		case subscriber.Receiver() <- msg:
```

Comment:
So it look like this bit now causes a race condition, see https://travis-ci.org/sensu/sensu-go/jobs/361886987.